package.json

```diff
@@ -79,8 +79,9 @@
     "eslint-plugin-import": "2.x",
     "in-publish": "^2.0.0",
     "linklocal": "^2.8.2",
-    "mocha": "^3.5.3",
-    "nyc": "^11.9.0",
+    "lodash.isarray": "^4.0.0",
+    "mocha": "^7.2.0",
+    "nyc": "^15.0.1",
```

Comment:
Unfortunately this is a change we can't ever really do; tests have to always be able to run in all the node versions we support, and we won't be dropping any node versions any time soon.

We can make the *build* require a newer node, and then run tests on the build output, as long as CI can be configured so it runs the build in newest node and then runs the tests in the appropriate node version.

Reply:
Neither the `build`-command, nor the `test`-command will work on windows if `nyc` is left on v11, like it is currently.

Reply:
What we could do instead would be to allow failures in appveyor tests.
Using `Windows Subsystem for Linux` we've got windows covered anyways.

Reply:
@ljharb what about the `dep-time-travel.sh`-script here?
https://github.com/benmosher/eslint-plugin-import/blob/b280acdce76cb209564451805687eb244ef9799d/tests/dep-time-travel.sh

Is there a chance we could use this to downgrade `nyc` and `mocha` before running tests?

Reply:
I'd be concerned about incpompatibilies across the versions; mocha and nyc config, for example, or the way coverage is calculated.

Unfortunately, WSL doesn't cover an actual windows filesystem, which is the point of the appveyor tests :-/

Again, in travis, this is solved with stages - I'm not sure how else to solve it for windows.

Perhaps it's worth moving Windows/WSL to Github Actions or to Travis?

Reply:
Currently I'm waiting for some AppVeyor tests to run. If the latest commit, which upgrades any existing `nyc` version to a predefined version, doesn't work, then we should rather move to GitHub Actions or Travis, yeah

Reply:
@ljharb alright, that's it 😄 
The "on the fly"-update of `nyc` works flawlessly now, all jobs passed! 🥳 
I'll just get rid of some errors (like peer-dependency stuff) now and then we're good to go, I think ^^

Reply:
Nvm - couldn't get rid of the peer-dependency error 😂
But still it's passing now ^^